libs/Microsoft.MixedReality.WebRTC.Native/include/interop_api.h

```diff
@@ -431,24 +594,77 @@ enum class VideoProfileKind : int32_t {
-MRS_API void MRS_CALL mrsPeerConnectionRegisterLocalAudioFrameCallback(
-    PeerConnectionHandle peerHandle,
-    PeerConnectionAudioFrameCallback callback,
-    void* user_data) noexcept;
+enum class mrsTransceiverStateUpdatedReason : int32_t {
+  kLocalDesc,
+  kRemoteDesc,
+  kSetDirection
+};
 
-/// Register a callback fired when an audio frame from an audio track was
-/// received from the remote peer.
-MRS_API void MRS_CALL mrsPeerConnectionRegisterRemoteAudioFrameCallback(
-    PeerConnectionHandle peerHandle,
-    PeerConnectionAudioFrameCallback callback,
-    void* user_data) noexcept;
+/// Flow direction of the media inside the transceiver. This maps to whether
+/// local and/or remote tracks are attached to the transceiver. The local
+/// track corresponds to the send direction, and the remote track to the
+/// receive direction.
+enum class mrsTransceiverDirection : int32_t {
+  kSendRecv = 0,
+  kSendOnly = 1,
+  kRecvOnly = 2,
+  kInactive = 3
+};
+
+/// Same as |mrsTransceiverDirection|, but including optional unset.
+enum class mrsTransceiverOptDirection : int32_t {
+  kNotSet = -1,
+  kSendRecv = 0,
+  kSendOnly = 1,
+  kRecvOnly = 2,
+  kInactive = 3
+};
+
+/// Configuration for creating a new audio transceiver.
+struct AudioTransceiverInitConfig {
+  /// Mandatory name of the audio transceiver. This must be a valid SDP token;
+  /// see |mrsSdpIsValidToken()|.
+  const char* name = nullptr;
+
+  /// Initial desired direction of the transceiver media when created.
+  mrsTransceiverDirection desired_direction =
+      mrsTransceiverDirection::kSendRecv;
+
+  /// Semi-colon separated list of stream IDs associated with the transceiver.
+  /// Use |Transceiver::DecodeStreamIDs()| and |Transceiver::EncodeStreamIDs()|.
+  const char* stream_ids = nullptr;
+
+  /// Handle of the audio transceiver interop wrapper, if any, which will be
+  /// associated with the native audio transceiver object.
+  mrsAudioTransceiverInteropHandle transceiver_interop_handle{};
+};
+
+/// Configuration for creating a new video transceiver.
+struct VideoTransceiverInitConfig {
+  /// Mandatory name of the video transceiver. This must be a valid SDP token;
+  /// see |mrsSdpIsValidToken()|.
+  const char* name = nullptr;
+
+  /// Initial desired direction of the transceiver media when created.
+  mrsTransceiverDirection desired_direction =
+      mrsTransceiverDirection::kSendRecv;
+
+  /// Semi-colon separated list of stream IDs associated with the transceiver.
+  /// Use |Transceiver::DecodeStreamIDs()| and |Transceiver::EncodeStreamIDs()|.
+  const char* stream_ids = nullptr;
+
+  /// Handle of the video transceiver interop wrapper, if any, which will be
+  /// associated with the native video transceiver object.
+  mrsVideoTransceiverInteropHandle transceiver_interop_handle{};
+};
+
+/// Configuration for opening a local audio capture device and creating a local
+/// audio track.
+struct LocalAudioTrackInitConfig {
```

Comment:
nit: move track name in here?
Move this to the local_audio_track header?

Reply:
Moved to local track header.